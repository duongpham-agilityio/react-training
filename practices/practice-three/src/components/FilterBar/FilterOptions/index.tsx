import { memo } from 'react';
import { Button, Link, List, ListItem } from '@chakra-ui/react';

// Mocks
import { filterOptions } from '@/mocks';

export const FilterOptions = memo(() => (
  <List display="flex" gap={5} py={5}>
    {filterOptions.map((option) => {
      const { name, value } = option;

      return (
        <ListItem key={value}>
          <Button
            as={Link} // Todo: Update to later
            href={`?filter_option=${value}`} // Todo: Update to later
            border="1px solid"
            borderColor="blackAlpha.300"
            borderRadius={25}
            boxShadow="md"
            bg="white"
            color="gray.20"
            fontWeight="regular"
            fontSize={16}
            px={15}
            _hover={{
              boxShadow: 'xl',
            }}
          >
            {name}
          </Button>
        </ListItem>
      );
    })}
  </List>
));
