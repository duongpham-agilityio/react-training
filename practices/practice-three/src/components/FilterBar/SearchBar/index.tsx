import { ChangeEvent, memo, useCallback } from 'react';
import { Center, Input, InputGroup } from '@chakra-ui/react';

// Icons
import { Search } from '@/assets/icons';

export type TSearchBarProps = {
  value: string;
  onChange: (value: string) => void;
};

export const SearchBar = memo(({ value, onChange }: TSearchBarProps) => {
  const handleChangeValue = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      const value = e.target.value;

      onChange(value);
    },
    [onChange],
  );

  return (
    <InputGroup
      border="1px"
      borderColor="gray.300"
      alignItems="center"
      borderRadius="md"
    >
      <Input
        fontSize="lg"
        placeholder="Search"
        borderColor="transparent"
        focusBorderColor="transparent"
        _hover={{
          borderColor: 'transparent',
        }}
        _placeholder={{
          fontWeight: 'light',
        }}
        value={value}
        onChange={handleChangeValue}
      />
      <Center h="60px" pr={5}>
        <Search />
      </Center>
    </InputGroup>
  );
});
