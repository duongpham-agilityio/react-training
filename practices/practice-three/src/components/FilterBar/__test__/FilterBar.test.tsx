import { render } from '@testing-library/react';

// Tests
import '@/components/FilterBar/__test__/SearchBar.test';
import '@/components/FilterBar/__test__/FilterOptions.test';

// Components
import { FilterBar, TFilterBarProps } from '..';

const onChange = jest.fn();
const defaultProps: TFilterBarProps = {
  value: '',
  onChange,
};
const setup = (props = defaultProps) => render(<FilterBar {...props} />);

describe('FilterBar', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
