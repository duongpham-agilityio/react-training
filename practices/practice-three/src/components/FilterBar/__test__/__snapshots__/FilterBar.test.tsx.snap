// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar Match to snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-1ey9w6j"
  >
    <div
      class="chakra-input__group css-y1t468"
      data-group="true"
    >
      <input
        class="chakra-input css-a7r4yx"
        placeholder="Search"
        value=""
      />
      <div
        class="css-1q3ec5g"
      >
        <svg
          fill="none"
          height="30"
          width="42"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M25.507 7.2H1.8a1.8 1.8 0 0 1 0-3.6h23.707a5.402 5.402 0 0 1 10.186 0H40.2a1.8 1.8 0 0 1 0 3.6h-4.507a5.402 5.402 0 0 1-10.186 0ZM40.2 26.4a1.8 1.8 0 0 0 0-3.6H16.493a5.402 5.402 0 0 0-10.186 0H1.8a1.8 1.8 0 0 0 0 3.6h4.507a5.402 5.402 0 0 0 10.186 0H40.2Z"
            fill="#6A6A6A"
          />
        </svg>
      </div>
    </div>
    <ul
      class="css-1c8ne0r"
      role="list"
    >
      <li
        class="css-0"
      >
        <a
          class="chakra-link chakra-button css-99bvj6"
          href="?filter_option=all"
        >
          All
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-link chakra-button css-99bvj6"
          href="?filter_option=fashion"
        >
          Fashion
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-link chakra-button css-99bvj6"
          href="?filter_option=electrics"
        >
          Electrics
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-link chakra-button css-99bvj6"
          href="?filter_option=books"
        >
          Books
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Filteroptions Match to snapshot 1`] = `
<div>
  <ul
    class="css-1c8ne0r"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        class="chakra-button css-16s5d4s"
        href="/?category=all"
      >
        All
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        class="chakra-button css-16s5d4s"
        href="/?category=fashion"
      >
        Fashion
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        class="chakra-button css-16s5d4s"
        href="/?category=electrics"
      >
        Electrics
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        class="chakra-button css-16s5d4s"
        href="/?category=books"
      >
        Books
      </a>
    </li>
  </ul>
</div>
`;

exports[`SearchBar Match to snapshot 1`] = `
<div>
  <div
    class="chakra-input__group css-viwvyh"
    data-group="true"
  >
    <input
      class="chakra-input css-190aajt"
      placeholder="Search"
      value=""
    />
    <div
      class="css-1q3ec5g"
    >
      <svg
        fill="none"
        height="30"
        width="42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M25.507 7.2H1.8a1.8 1.8 0 0 1 0-3.6h23.707a5.402 5.402 0 0 1 10.186 0H40.2a1.8 1.8 0 0 1 0 3.6h-4.507a5.402 5.402 0 0 1-10.186 0ZM40.2 26.4a1.8 1.8 0 0 0 0-3.6H16.493a5.402 5.402 0 0 0-10.186 0H1.8a1.8 1.8 0 0 0 0 3.6h4.507a5.402 5.402 0 0 0 10.186 0H40.2Z"
          fill="#6A6A6A"
        />
      </svg>
    </div>
  </div>
</div>
`;
